### DISCUSSION TITLE
run cnv_end2end_example.ipynb have mistake

### DISCUSSION THREAD

**Discussion by Ajfil** [Xilinx/finn#780] [Q&A] [answered]
when i execute Hardware Generation,there are mistake :AttributeError: module 'numpy' has no attribute 'str'.
`np.str` was a deprecated alias for the builtin `str`. To avoid this error in existing code, use `str` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.str_` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
i think is porblem of numpy version,my numpy version is 1.24.2,but i build environment by throughing run-docker.sh,my english is not good ,I hope someone can help me with this problem,thank you so much!!!
![K`ISYP%3}NQ E{V 7~XQH D](https://user-images.githubusercontent.com/91005367/222959445-131f8ab5-c7af-469d-882e-0a15866f5aef.png)

**Comment by auphelia** [comment, marked as answer]
Hi @Ajfil, I think you're working with an older FINN version. Please checkout the new FINN release (v0.9), this was already solved in this commit: https://github.com/Xilinx/finn/commit/1b8bf757c112285c1f98b60ba2d475c53643aa6a
